pkg/controller/dgraph/models/query/metrics.go

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright (c) 2018 VMware Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package query
+
+import (
+	"github.com/vmware/purser/pkg/controller/dgraph/models"
+)
+
+// DaemonsetMetrics query
+func getQueryForDaemonsetMetrics(name string) string {
+	return `query {
+		parent(func: has(isDaemonset)) @filter(eq(name, "` + name + `")) {
+			children: ~daemonset @filter(has(isPod)) {
+				` + getQueryForMetricsComputationWithAliasAndVariables("Pod") + `
+			}
+			` + getQueryForAggregatingChildMetricsWithAlias("Pod") + `
+		}
+	}`
+}
+
+// JobMetrics query
+func getQueryForJobMetrics(name string) string {
+	return `query {
+		parent(func: has(isJob)) @filter(eq(name, "` + name + `")) {
+			children: ~job @filter(has(isPod)) {
+				` + getQueryForMetricsComputationWithAliasAndVariables("Pod") + `
+			}
+			` + getQueryForAggregatingChildMetricsWithAlias("Pod") + `
+		}
+	}`
+}
+
+// ReplicasetMetrics query
+func getQueryForReplicasetMetrics(name string) string {
+	return `query {
+		parent(func: has(isReplicaset)) @filter(eq(name, "` + name + `")) {
+			children: ~replicaset @filter(has(isPod)) {
+				` + getQueryForMetricsComputationWithAliasAndVariables("Pod") + `
+			}
+			` + getQueryForAggregatingChildMetricsWithAlias("Pod") + `
+		}
+	}`
+}
+
+// StatefulsetMetrics query
+func getQueryForStatefulsetMetrics(name string) string {
+	return `query {
+		parent(func: has(isStatefulset)) @filter(eq(name, "` + name + `")) {
+			children: ~statefulset @filter(has(isPod)) {
+				` + getQueryForMetricsComputationWithAliasAndVariables("Pod") + `
+			}
+			` + getQueryForAggregatingChildMetricsWithAlias("Pod") + `
+		}
+	}`
+}
```

Comment:
These 4 methods are identical, let's abstract these as well to a common method such as `getQueryForResourceMetrics()` with arguments for the variables.

Reply:
Done